eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs

```diff
@@ -604,3 +640,71 @@ instance
   DecCBOR (Annotator (AlonzoTxBodyRaw era))
   where
   decCBOR = pure <$> decCBOR
+
+alonzoRedeemerPointer ::
+  forall era.
+  MaryEraTxBody era =>
+  TxBody era ->
+  AlonzoPlutusPurpose AsItem era ->
+  StrictMaybe (AlonzoPlutusPurpose AsIndex era)
+alonzoRedeemerPointer txBody = \case
+  AlonzoSpending txIn ->
+    AlonzoSpending <$> indexOf txIn (txBody ^. inputsTxBodyL)
+  AlonzoMinting policyID ->
+    AlonzoMinting <$> indexOf policyID (txBody ^. mintedTxBodyF :: Set (PolicyID (EraCrypto era)))
```

Comment:
Should we switch the type of `mintedTxBody` to be an `OSet`?

Reply:
That would be great, but we can't do that right now, unfortunately
We would have to change `mintTxBodyF` to an `OMap` first.